src/App.js

```diff
@@ -43,28 +45,28 @@ class App extends Component {
         super();
         this.state = {
             cData: {},
-            currentLang: chartData[0][0].name,
+            currentLang: chartData[currentCatIndexGlobal][0].name,
             /*
             currentCatIndex :-
                 Web---------------------> 0
                 Mobile------------------> 1
                 Programming Language----> 2
                 Backend-----------------> 3
             */
-            currentCatIndex: 0,
-            arrObj: dataExtractor()
+            arrObj: dataExtractor(0)
         }
     }
 
     componentDidMount() {
         this.getData(this.state.currentLang);
     }
 
-    getData(currentSelection) {
-        const {langArray, devLoveArray, gJobArray, usJobArray, supJobArray, remJobArray} = this.state.arrObj;
+    getData(currentSelection, object) {
```

Comment:
where is this object coming from? You should name it better so we understand what it is

Reply:
The dataExtractor function returns an object..

Reply:
in the code above, it receives an object as a parameter. It should be receiving the state object. `object` isn't anything right now in the code

Reply:
what i mean is: rename object to be something more descriptive so we know what it receives. Also how come line 91 doesn't receive an object as a parameter?

Reply:
The dataExtractor function returns an object which i have assigned to a variable "obj"
Now, when this.getData() is called i have passed the above "obj" as an argument
And the function getData will receive this as a parameter..

Reply:
In line 91 we are calling this.getData function

Reply:
I understand how the function works. I am simply asking to rename obj to something more descriptive like chartData

Line 91 is only receiving one parameter instead of 2. We should keep things consistent and make sure we are always calling the function the same way

Reply:
Sorry Andrei, i was not trying to explain you. And yes i will definitely change the name to something better..

Reply:
:)

Reply:
👍 :) no problem at all!